src/pages/ExamsPage/components/AllowanceListActions.jsx

```diff
@@ -10,8 +11,106 @@ import {
 import { DeleteOutline, EditOutline } from '@openedx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
 
+import React, { useState } from 'react';
 import messages from '../messages';
-import { useDeleteAllowance } from '../hooks';
+import { useDeleteAllowance, useEditAllowance, useAllowancesData } from '../hooks';
+
+const EditModal = (isOpen, onClose, allowance, formatMessage) => {
+  const editAllowance = useEditAllowance();
+  const {
+    allowancesList,
+  } = useAllowancesData();
+
+  const allowances = allowancesList.filter(a => a.id === allowance.id);
+  const [additionalTimeError, setAdditionalTimeError] = useState(false);
+
+  const initialFormState = {};
+  const [form, setForm] = useState(initialFormState);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setForm(prev => ({ ...prev, [name]: value }));
+  };
+
+  const onSubmit = () => {
+    // todo: make sure that validation is correct
+    setAdditionalTimeError(!form['extra-time-mins']);
+    const valid = (
+      form['extra-time-mins']
+    );
+    if (valid) {
+      form['extra-time-mins'] = Number(form['extra-time-mins']);
+      const payload = {
+        username: allowances[0].username,
+        exam_id: allowance.examId,
+        extra_time_mins: form['extra-time-mins'],
```

Comment:
Is it okay to have the payload/form data added like this?

Reply:
IMHO, We should move the value to its own variable to cast and validate and then add it to the payload rather than mutating the form. The form reference is obtained from the state and mutating it might cause side effects.

I recommend something like my suggestion above of casting it as:
```javascript
const extraTimeMins  = +form['extra-time-mins'] || 0;
```

And then if everything is ok then use it as:
```javascript
const payload = {
  (...)
  extra_time_mins: extraTimeMins,
}
```

Reply:
I think your idea of passing the values `username`, `exam_id`, and `extra_time_mins` to `editAllowance` is reasonable. That way, the hook is responsible for crafting the request body instead of the component, which can be more focused on display. On the other hand, that's not consistent with the way the rest of the repository works, so it may be better to keep consistency, especially since `payload` matches the shape of `initialForm` anyway, so you can't really decouple the form shape and the payload shape.

Reply:
That makes sense around consistency. I will keep as is, Michael. And Marcos, good call about setting the variable and casting rather than mutating the form. Thanks!